shiny/experimental/ui/_utils.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+import typing
+
+from htmltools import TagAttrs, TagAttrValue, TagChild, div
+
+from ..._typing_extensions import TypeGuard
+
+T = typing.TypeVar("T", bound=TagChild)
+
+# Export this type from htmltools
+TagAttrDict = typing.Dict[str, str]
+
+
+def consolidate_attrs(
+    *args: T | TagAttrs,
+    **kwargs: TagAttrValue,
+) -> tuple[TagAttrDict, list[T]]:
+    tag = div(*args, **kwargs)
+    # While returning `tag.children` works, it is nice to have a minimal type hint
+    children = typing.cast(typing.List[T], tag.children)
```

Comment:
Either:

- Don't cast, `children` should be a `TagList`, or
- construct the `children` yourself in a way that fits `list[T]`

Reply:
Now casting `attrs` to `TagAttrs` as `TagAttrDict` is not compatible.

Also constructing `list[T]` the same way `Tag` does before sending the _children_ to `TagList()`